src/spikeinterface/generation/session_displacement_generator.py

```diff
@@ -0,0 +1,510 @@
+    output_sortings : list
+        A list of corresponding sorting objects.
+    extra_outputs_dict (options) : dict
+        When `extra_outputs` is `True`, a dict containing variables used
+        in the generation process.
+        "unit_locations" : A list (length num records) of shifted unit locations
+        "templates_array_moved" : list[np.array]
+            A list (length num records) of (num_units, num_samples, num_channels)
+            arrays of templates that have been shifted.
+
+    Notes
+    -----
+    It is important to consider what unit properties are maintained
+    across the session. Here, all `generate_template_kwargs` are fixed
+    across sessions, to be sure the unit properties do not change.
+    The firing rates passed to `generate_sorting` for each unit are
+    also fixed across sessions. When a seed is set, the exact spike times
+    will also be fixed across recordings. otherwise, when seed is `None`
+    the actual spike times will be different across recordings, although
+    all other unit properties will be maintained (except any location
+    shifting and template scaling applied).
+    """
+    # temporary fix
+    generate_unit_locations_kwargs = copy.deepcopy(generate_unit_locations_kwargs)
+    generate_templates_kwargs = copy.deepcopy(generate_templates_kwargs)
+    generate_sorting_kwargs = copy.deepcopy(generate_sorting_kwargs)
+    generate_noise_kwargs = copy.deepcopy(generate_noise_kwargs)
+
+    _check_generate_session_displacement_arguments(
+        num_units, recording_durations, recording_shifts, recording_amplitude_scalings, shift_units_outside_probe
+    )
+
+    probe = generate_probe(generate_probe_kwargs, probe_name)
+    channel_locations = probe.contact_positions
+
+    # Create the starting unit locations (which will be shifted).
+    unit_locations = generate_unit_locations(
+        num_units,
+        channel_locations,
+        seed=seed,
+        **generate_unit_locations_kwargs,
+    )
+
+    # Fix generate template kwargs, so they are the same for every created recording.
+    # Also fix unit firing rates across recordings.
+    fixed_generate_templates_kwargs = fix_generate_templates_kwargs(generate_templates_kwargs, num_units, seed)
+
+    fixed_firing_rates = _ensure_firing_rates(generate_sorting_kwargs["firing_rates"], num_units, seed)
+    fixed_generate_sorting_kwargs = copy.deepcopy(generate_sorting_kwargs)
+    fixed_generate_sorting_kwargs["firing_rates"] = fixed_firing_rates
+
+    if shift_units_outside_probe:
+        # Create a new set of templates one probe-width above and
+        # one probe-width below the original templates. The number of
+        # units is duplicated for each section, so the new num units
+        # is 3x the old num units.
+        num_units, unit_locations, fixed_generate_templates_kwargs, fixed_generate_sorting_kwargs = (
+            _update_kwargs_for_extended_units(
```

Comment:
i guess it would only need max_displacement padding of units? maybe plus a spread factor of 200um?

also, would it be worth adding this to the single-session generator too?